x/nft/codec.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+
+package nft;
+
+message NonFungibleToken {
+    bytes id = 1;
+    bytes owner = 2;
+    repeated ActionApprovals actionApprovals = 3;
+
+}
+
+message ActionApprovals {
+    string action = 1;
+    repeated Approval approvals = 2;
+}
+
+message Approval {
+    bytes address = 1;
+    ApprovalOptions options = 2;
+}
+
+message ApprovalOptions {
```

Comment:
:+1:

default values `(0, 0, false)` means never expire, but erased on transfer of ownership? Seems like a good default